plugin/code_actions.py

```diff
@@ -80,23 +79,7 @@ class CodeActionsManager:
     """Manager for per-location caching of code action responses."""
 
     def __init__(self) -> None:
-        self._point_cache = {}  # type: Dict[str, CodeActionsCollector]
-
-    def request_for_point(
-        self,
-        view: sublime.View,
-        actions_handler: Callable[[CodeActionsByConfigName], None],
-        request_point: int
-    ) -> None:
-        current_location_key = "{}#{}:{}".format(view.buffer_id(), view.change_count(), request_point)
-        if current_location_key in self._point_cache:
-            actions_handler(self._point_cache[current_location_key].get_actions())
-        else:
-            self._point_cache.clear()
-            point = offset_to_point(view, request_point)
-            diagnostics_by_config, extended_range = filter_by_point(view_diagnostics(view), point)
-            results = self.request_with_diagnostics(view, extended_range, diagnostics_by_config, actions_handler)
-            self._point_cache[current_location_key] = results
+        self._response_cache = {}  # type: Dict[str, CodeActionsCollector]
```

Comment:
Remind me again when items from this dict are popped? Or is it growing to infinity?

Reply:
It's cleared on every request that doesn't match the cache: https://github.com/sublimelsp/LSP/blob/06a59d66e904d522b29209a8b42d4deb3aa296bb/plugin/code_actions.py#L139-L140

The fact that it's a dict is maybe a bit confusing here. Could be a tuple of cache_key, response.

Reply:
I'd much prefer that. Dicts can grow indefinitely, tuples cannot :)